app/controllers/OwnerApi.scala

```diff
@@ -2,64 +2,64 @@ package controllers
 
 import agent.{Label, Origin, ResourceType}
 import data.Owners
+import javax.inject.Inject
 import jsonimplicits.model._
 import model.Owner
-import org.joda.time.{DateTime, Duration}
+import org.joda.time.DateTime
+import play.api.mvc._
+import play.api.mvc.{Action, RequestHeader, Result}
 import play.api.libs.json.{JsObject, Json, Writes}
 import play.api.libs.json.Json._
-import play.api.mvc.{Action, Controller, RequestHeader, Result}
-import utils.ResourceFilter
-
+import scala.concurrent.ExecutionContext
 import scala.concurrent.Future
 import scala.concurrent.duration._
 import scala.language.postfixOps
+import utils.ResourceFilter
 
+class OwnerApi @Inject()(cc: ControllerComponents, ec: ExecutionContext) extends AbstractController(cc) {
 
-object OwnerApi extends Controller {
-
-  private def itemsResult[T](key: String, items: => Iterable[T], baseObject: JsObject = Json.obj())(implicit request: RequestHeader, tw: Writes[T]): Future[Result] = {
-    ApiResult.filter {
-      Map(ownerLabel -> items.toSeq)
-    } reduce { collection =>
-      baseObject ++ Json.obj(key -> toJson(collection.values.flatten))
+    private def itemsResult[T](key: String, items: => Iterable[T], baseObject: JsObject = Json.obj())(implicit request: RequestHeader, tw: Writes[T]): Future[Result] = {
+      ApiResult.filter {
+        Map(ownerLabel -> items.toSeq)
+      } reduce({ collection =>
+        baseObject ++ Json.obj(key -> toJson(collection.values.flatten))
+      }, ec)
     }
-  }
 
-  object OwnerResourceType extends ResourceType("Owners", 30 days, 30 days)
+    object OwnerResourceType extends ResourceType("Owners", 30 days, 30 days)
 
-  val ownerLabel = Label(
-    OwnerResourceType,
-    new Origin {
-      val vendor = "prism"
-      val account = "prism"
-      val resources = Set("sources")
-      val jsonFields = Map.empty[String, String]
-    },
-    Owners.all.size,
-    DateTime.now
-  )
+    val ownerLabel: Label = Label(
+      OwnerResourceType,
+      new Origin {
+        val vendor = "prism"
+        val account = "prism"
+        val resources = Set("sources")
+        val jsonFields = Map.empty[String, String]
+      },
+      Owners.all.size,
+      DateTime.now
+    )
 
-  def ownerList = Action.async { implicit request =>
-    val requestFilter =  ResourceFilter.fromRequestWithDefaults()
-    def filter(owner: Owner) = {
-      val json = Json.toJson(owner)
-      if (requestFilter.isMatch(json)) Some(json) else None
+    def ownerList: Action[AnyContent] = Action.async { implicit request =>
```

Comment:
Same thing on messy types here. I'd disable the intention for controllers.

Reply:
🛠️